### PR TITLE
Import Elixir SDK as Experimental

### PR DESCRIPTION
**Pull request by gerhard**
This imports https://github.com/wingyplus/dagger_ex under `sdk/elixir` as an **experimental** SDK.

## @gerhard action items (part of this PR)
- [x] Open PR with https://github.com/wingyplus/dagger_ex history import
	- [x] Capture commands used for history import
```console
git remote add dagger-sdk-elixir git@github.com:wingyplus/dagger_ex
git fetch dagger-sdk-elixir --no-tags
git checkout -b sdk/elixir dagger-sdk-elixir/maim
mkdir -p sdk/elixir
git ls-tree -z --name-only HEAD | xargs -0 -I {} git mv {} sdk/elixir/
git commit --gpg-sign --signoff

git checkout main
git pull
git checkout -b import-sdk-elixir

git merge --allow-unrelated-histories sdk/elixir

git push -u fork
```
- [x] Give @wingyplus write permissions to `dagger` repository
- [x] Add @wingyplus to [`CODEOWNERS`](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) (`@dagger/sdk-elixir` group)
  - If we like this, I propose that we do the same for other SDKs cc @shykes @samalba @aluzzardi 
- [x] Mention in SDK's README that Elixir is currently experimental

## @wingyplus action items (part of this PR)
- [x] Configure `Sign-off-by` https://docs.dagger.io/204441/contributing/#dco
  - @gerhard manually set this to **Pass** so that @wingyplus does not need to rewrite `255` commits
- [x] Move (and rename) GHA workflows from `sdk/elixir/.github` to `.github/`
- [x] Ensure all GHA workflows pass

## Follow-ups (outside / after this PR)
This sounds like maintainer on-boarding to me @kjuulh 🙂

- Migrate [open issues](https://github.com/wingyplus/dagger_ex/issues) which are still relevant - `10` open currently
	- Ensure that we have a new `sdk/elixir` label cc @d3rp3tt3 @mircubed
- Migrate PRs which are still relevant
- Add commit to https://github.com/wingyplus/dagger_ex that removes all files except the README
	- Add repo migration notice to `README.md`
	- Archive repository in GitHub, e.g. https://github.com/thechangelog/infrastructure
- Tag `sdk/elixir/v*` to check that a new release gets created as expected
  - This will require us to check that all three crates are published `by Dagger`. FTR: https://www.hex.pm/packages/dagger_ex - **maybe** rename to https://www.hex.pm/packages/dagger
- Add Elixir SDK to https://github.com/dagger/dagger/blob/main/RELEASING.md
  - cc @gerhard 
- Integrate with Analytics
  - cc @grouville @d3rp3tt3 

Did I miss anything @shykes @samalba @wingyplus @d3rp3tt3 @mircubed?

### CODE REVIEW

**Comment by kjuulh** [comment]
Looks good from my pov. I guess the next steps would be adopting the hack/ scripts and integrating ci?

**Comment by kjuulh** [comment]
FYI I am not quite sure why I am tagged xD

**Comment by wingyplus** [comment]
Note for myself:

- [ ] Fix tests failure 
- [ ] Fix `mix dagger.ci` to return the proper exit code. Currently, it still passes even tests that are failures.
- [ ] Integrate ci to dagger. Never looked in the dagger repo but sounds fun. :D

**Comment by wingyplus** [comment]
Hi @gerhard, 

I've done porting `sdk/elixir/.github` into `.github` and ported the test and lint pipeline into `internal/mage/sdk` already. I'm not sure about 3 mage tasks left (generate, bump, publish), do I need to tackle in this PR or do it later?

File: sdk/elixir/mix.exs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  defp package do
+    %{
+      name: "dagger_ex",
+      description: "Dagger SDK for Elixir",
+      licenses: ["MIT"],
+      links: %{
+        "GitHub" => "https://github.com/dagger/dagger",
+        "Changelog" => "#{@source_url}/blob/sdk/elixir/v#{@version}/CHANGELOG.md"
```
Comment (wingyplus) [inline sdk/elixir/mix.exs:41]: This is just a link that shows on https://hex.pm/packages/dagger_ex. We can use a release note from GitHub release page instead.

**Comment by gerhard** [comment]
> @kjuulh FYI I am not quite sure why I am tagged xD

My mistake, I removed you from the discussion 👍

**Comment by gerhard** [comment]
> I've done porting `sdk/elixir/.github` into `.github` and ported the test and lint pipeline into `internal/mage/sdk` already.

🙌 will be checking them out shortly

> I'm not sure about 3 mage tasks left (~generate~, bump, publish), do I need to tackle in this PR or do it later?

All of these can be done in follow-up PRs, no worries.

**Comment by gerhard** [comment]
All the changes look good to me.

Would you like to rebase everything on top of `main` before we merge @wingyplus?

After you do that, if everything looks good to you, please approve the PR @wingyplus so that we can get it merged 💪

**Review by wingyplus — APPROVED** [review]

**Comment by wingyplus** [comment]
@gerhard The PR is now rebased. And all tests related to Elixir are now passed except `test / engine` which seems not related to this PR. 🙇

**Comment by wingyplus** [comment]
Rebased

**Comment by wingyplus** [comment]
I found lint-daggerized workflow error due to pointing to incorrect local action (ref https://github.com/dagger/dagger/actions/runs/5147441125). I sent a fix here https://github.com/dagger/dagger/pull/5244.

**Pull request merged**